api/src/routes/certificate.test.ts

```diff
@@ -0,0 +1,336 @@
+            isApisMicroservicesCert: true,
+            isQaCertV7: true,
+            isSciCompPyCertV7: true,
+            isDataAnalysisPyCertV7: true,
+            isInfosecCertV7: true,
+            isMachineLearningPyCertV7: true,
+            isCollegeAlgebraPyCertV8: true,
+            isFoundationalCSharpCertV8: true
+          }
+        });
+
+        const spy = jest.spyOn(fastifyTestInstance, 'sendEmail');
+
+        const response = await superRequest('/certificate/verify', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          certSlug: Certification.RespWebDesign
+        });
+
+        expect(spy).toHaveBeenCalled();
+        expect(response.status).toBe(200);
+      });
+
+      test('should return 200 if all went well', async () => {
+        await fastifyTestInstance.prisma.user.updateMany({
+          where: { email: defaultUserEmail },
+          data: {
+            completedChallenges: [
+              { id: 'bd7158d8c442eddfaeb5bd18', completedDate: 123456789 },
+              { id: '587d78af367417b2b2512b03', completedDate: 123456789 },
+              { id: '587d78af367417b2b2512b04', completedDate: 123456789 },
+              { id: '587d78b0367417b2b2512b05', completedDate: 123456789 },
+              { id: 'bd7158d8c242eddfaeb5bd13', completedDate: 123456789 }
+            ],
+            isRespWebDesignCert: false
+          }
+        });
+
+        const response = await superRequest('/certificate/verify', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          certSlug: Certification.RespWebDesign
+        });
+
+        const user = await fastifyTestInstance.prisma.user.findFirst({
+          where: { email: defaultUserEmail }
+        });
+
+        expect(user).toMatchObject({ isRespWebDesignCert: true });
+        expect(response.body).toMatchObject({
+          response: { message: 'flash.cert-claim-success' }
+        });
+        expect(response.status).toBe(200);
+      });
+
+      // Tests for all certifications as to what may currently be claimed, and what may no longer be claimed
+      test('should return 400 if certSlug is not allowed', async () => {
+        const claimableCerts = [
+          Certification.RespWebDesign,
+          Certification.JsAlgoDataStruct,
+          Certification.FrontEndDevLibs,
+          Certification.DataVis,
+          Certification.RelationalDb,
+          Certification.BackEndDevApis,
+          Certification.QualityAssurance,
+          Certification.SciCompPy,
+          Certification.DataAnalysisPy,
+          Certification.InfoSec,
+          Certification.MachineLearningPy,
+          Certification.CollegeAlgebraPy,
+          Certification.FoundationalCSharp,
+          Certification.LegacyFrontEnd,
+          Certification.LegacyBackEnd,
+          Certification.LegacyDataVis,
+          Certification.LegacyInfoSecQa,
+          Certification.LegacyFullStack
+        ];
+        const unclaimableCerts = [];
```

Comment:
```suggestion
        const unclaimableCerts = ['fake-slug'];
```
So we always test the branch where the slug is invalid.